en/guide/sdk.md

```diff
@@ -0,0 +1,155 @@
+build them into DroneCore at compile time. <!-- It also provides other functionality that is useful for SDK developers,
+including the ability to lock down custom versions of DroneCore to authorised developers. -->
+
+> **Note** This topic only covers the (very minor) differences between SDK plugins/testing and "vanilla" DroneCore.
+> SDK authors should review the topics [Writing Plugins](../contributing/plugins.md) and 
+> [Testing](../contributing/test.md).
+
+
+## SDK Architecture
+
+SDK extensions are defined in a separate/parallel folder to DroneCore. 
+The extension folder *mirrors* DroneCore, and **must** contain the folders **integration_tests** and **plugins**.
+The *plugin* folder contains appropriately named sub-folders for each plugin (along with its unit test code).
+
+A simplified view of a "typical" DroneCore SDK is shown below (with extension directory: *SDK_Extensions*). 
+
+```
+├── DroneCore
+│   ├── core
+│   ├── integration_tests
+│   └── plugins
+│       ├── action
+│       ├── ...
+│       └── telemetry
+├── SDK_Extensions
+│   ├── integration_tests
+│   └── plugins
+│       ├── camera
+│       ├── another_plugin
+│       └── etc.
+```
+
+
+## Create an SDK Extension Library
+
+To create a new C++ SDK Extension Library, first copy the [DroneCore/external_example](https://github.com/dronecore/DroneCore/tree/master/external_example) 
+alongside DroneCore. 
+
+The *external_example* contains a single integration test (**hello_world.cpp**) and a single plugin (**/example**) 
+with a separate implementation file and a stub unit test file. Adapt it first by renaming the top level folder.
+```
+external_example
+├── integration_tests
+│   ├── CMakeLists.txt
+│   └── hello_world.cpp
+└── plugins
+    └── example
+        ├── CMakeLists.txt
+        ├── example.cpp
+        ├── example.h
+        ├── example_impl.cpp
+        ├── example_impl.h
+        └── example_impl_test.cpp
+```
+
+## Adding/Modifying Plugins
+
+The extension library can contain as many plugins as needed, provided they are subfolders of **/plugins**, 
+and have an appropriately formatted **CMakeLists.txt** make file. You can therefore duplicate and modify 
+the "example" plugin as required by the SDK.
+
+Modify the example plugin as needed and update its [CMakeLists.txt](https://github.com/dronecore/DroneCore/blob/master/external_example/plugins/example/CMakeLists.txt) as appropriate:
+* Add additional libraries using the variable `additional_libs`:
+  ```
+  set(additional_libs "library_name" PARENT_SCOPE)
+  ```
+* Add required includes with `additional_includes`:
+  ```
+  set(additional_includes "include_dir" PARENT_SCOPE)
+  ```
+* You can also add tests with `unittest_source_files`, as discussed in the following section.
+ 
+Plugins are otherwise written exactly as discussed in [Writing Plugins](../contributing/plugins.md). 
+
+
+## Testing
+
+SDK Extension tests are the same as those for the default plugins. 

```

Comment:
@julianoes On reflection I think that tests are the same for all types of plugins - just defined in different CMakeList.txt, so this can be cut back and just point to the test docs (which may be then moved to the plugin docs)